pandas/util/testing.py

```diff
@@ -587,19 +593,20 @@ def _get_ilevel_values(index, level):
         raise_assert_detail(obj, msg1, msg2, msg3)
 
     # MultiIndex special comparison for little-friendly error messages
-    if left.nlevels > 1:
-        for level in range(left.nlevels):
-            # cannot use get_level_values here because it can change dtype
-            llevel = _get_ilevel_values(left, level)
-            rlevel = _get_ilevel_values(right, level)
-
-            lobj = 'MultiIndex level [{level}]'.format(level=level)
-            assert_index_equal(llevel, rlevel,
-                               exact=exact, check_names=check_names,
-                               check_less_precise=check_less_precise,
-                               check_exact=check_exact, obj=lobj)
-            # get_level_values may change dtype
-            _check_types(left.levels[level], right.levels[level], obj=obj)
+    if isinstance(left, MultiIndex) and isinstance(right, MultiIndex):
```

Comment:
this is needed to avoid type issues?

Reply:
Yes. Otherwise, mypy showed
`pandas/util/testing.py:608: error: "Index" has no attribute "levels"; maybe "nlevels"?`

Reply:
Shouldn't this just replace the existing condition then? Seems duplicative to have both checks